### ISSUE TITLE
ggv2setup_test: TestScenerios Test Flake:  expected 1 listeners, got 0

### ISSUE THREAD

**Issue by sam-heilbron** [k8sgateway/k8sgateway#10375]
### Which tests failed?

Error:
```
 --- FAIL: TestScenarios/failover-default-diffns (1.20s)
        ggv2setup_test.go:276: applied yamls TestScenarios/failover-default-diffns
        ggv2setup_test.go:349: expected 1 listeners, got 0
        ggv2setup_test.go:355: expected 1 routes, got 0
        ggv2setup_test.go:378: listener http not found
        ggv2setup_test.go:381: listener http not equal
        ggv2setup_test.go:391: route http-routes-14610285773269467959 not found
        ggv2setup_test.go:394: route http-routes-14610285773269467959 not equal: name:"http-routes-14610285773269467959"  virtual_hosts:{name:"http~www_example_com"  domains:"www.example.com"  routes:{name:"http~www_example_com-route-0-httproute-http-gwtest-0-0-matcher-0"  match:{prefix:"/"}  route:{cluster:"kube-svc:gwtest-reviews-8080_gwtest"  cluster_not_found_response_code:INTERNAL_SERVER_ERROR}}}  ignore_port_in_host_matching:true vs <nil>
        ggv2setup_test.go:271: deleted yamls TestScenarios/failover-default-diffns
    ggv2setup_test.go:80: 2024-11-22T01:44:20.768Z	DEBUG	k8s.k8s-gw-proxy-syncer	proxy_syncer/proxy_syncer.go:421	building proxy for kube gw gwtest/http-failover-default version 381
```

Logs: https://github.com/solo-io/gloo/actions/runs/11964832927/job/33357822377?pr=10393

**Comment by danehans** [comment]
The key error:
```sh
ERROR controller/controller.go:316 Reconciler error {"controller": "gateway", ..., "error": "failed to render helm chart for gateway gwtest.http-failover-default-diffns: template: gloo-gateway/templates/gateway/proxy-deployment.yaml:115:65: executing \"gloo-gateway/templates/gateway/proxy-deployment.yaml\" at <$gateway.sdsContainer.image>: nil pointer evaluating interface {}.image"}
```
The test is failing because the Gloo Gateway controller encounters an error while rendering the Helm chart for the Gateway resource `gwtest.http-failover-default-diffns`. I have hit a similar error and resolved it by recreating the expected test output files:

1. `rm -rf projects/gateway2/setup/testdata/*-out.yaml`
2. Rerunning the setup tests twice (the first time fails but recreates the output files).

c: @yuval-k

**Comment by danehans** [comment]
I just ran the tests locally without hitting this issue:

```sh
$ go test -v ./projects/gateway2/setup
...
--- PASS: TestScenarios (31.74s)
    --- PASS: TestScenarios/failover-default-diffns (1.14s)
    --- PASS: TestScenarios/failover-default (1.10s)
    --- PASS: TestScenarios/failover (1.16s)
    --- PASS: TestScenarios/happypath (1.10s)
    --- PASS: TestScenarios/listeneropts (1.08s)
PASS
ok  	github.com/solo-io/gloo/projects/gateway2/setup	31.809s
```

**Comment by sam-heilbron** [comment]
We expect that this was resolved by https://github.com/solo-io/gloo/pull/10410. As a result, this is fixed on the 1.18 and main branches, which we believe are the only two branches impacted.
